modules/fedimint-dummy-common/src/db.rs

```diff
@@ -1,64 +1,107 @@
 use fedimint_core::db::DatabaseTransaction;
 use fedimint_core::encoding::{Decodable, Encodable};
-use fedimint_core::{impl_db_lookup, impl_db_record};
+use fedimint_core::{impl_db_lookup, impl_db_record, Amount, OutPoint};
 use futures::StreamExt;
+use secp256k1::XOnlyPublicKey;
 use serde::Serialize;
 use strum_macros::EnumIter;
 
-/// Example function that will migrate the Dummy Module's database from
-/// version 0 to version 1. This function selects all of the ExampleKeyV0
-/// and inserts a new String to construct ExampleKeys, deletes the old
-/// ExampleKeyV0, then inserts the new ExampleKeys.
-pub async fn migrate_dummy_db_version_0<'a, 'b>(
-    dbtx: &'b mut DatabaseTransaction<'a>,
-) -> Result<(), anyhow::Error> {
-    let example_keys_v0 = dbtx
-        .find_by_prefix(&ExampleKeyPrefixV0)
-        .await
-        .collect::<Vec<_>>()
-        .await;
-    dbtx.remove_by_prefix(&ExampleKeyPrefixV0).await;
-    for (key, val) in example_keys_v0 {
-        let key_v2 = ExampleKey(key.0, "Example String".to_string());
-        dbtx.insert_new_entry(&key_v2, &val).await;
-    }
-    Ok(())
-}
+use crate::{DummyOutputOutcome, DummyPrintMoneyRequest};
 
+/// Namespaces DB keys for this module
 #[repr(u8)]
 #[derive(Clone, EnumIter, Debug)]
 pub enum DbKeyPrefix {
```

Comment:
Nit: not a big deal for the dummy module but I think its a bit weird to have a shared db.rs for the client/server since I don't think they shouldn't ever access the same database structures

Reply:
Was following a pattern in the other modules, but I think you are right.